### ISSUE TITLE
libav build issue on Ubuntu 14.04 LTS

### ISSUE THREAD

**Issue by ghost** [qTox/qTox#2027]
Hi,
I  am trying to build qTox from source.

Here's the build output:

```
maxi@tuxedo:~/git/qTox$ make
g++ -c -m64 -pipe -fno-exceptions -O2 -std=c++0x -Wall -W -D_REENTRANT -fPIE -DGIT_VERSION="\"e3dd2dc9e11ff5208e6ead39139c64e0f20db6d9\"" -DTIMESTAMP=1437493560 -DLOG_TO_FILE -DQT_MESSAGELOGCONTEXT -DQTOX_PLATFORM_EXT -DQTOX_FILTER_AUDIO -DENABLE_SYSTRAY_STATUSNOTIFIER_BACKEND -DENABLE_SYSTRAY_GTK_BACKEND -DQT_NO_DEBUG -DQT_SVG_LIB -DQT_OPENGL_LIB -DQT_WIDGETS_LIB -DQT_SQL_LIB -DQT_XML_LIB -DQT_NETWORK_LIB -DQT_GUI_LIB -DQT_CORE_LIB -I/usr/lib/x86_64-linux-gnu/qt5/mkspecs/linux-g++-64 -I. -Ilibs/include -I/usr/include/glib-2.0 -I/usr/include/gtk-2.0 -I/usr/include/atk-1.0 -I/usr/include/cairo -I/usr/include/gdk-pixbuf-2.0 -I/usr/include/pango-1.0 -I/usr/lib64/glib-2.0/include -I/usr/lib/x86_64-linux-gnu/glib-2.0/include -I/usr/lib64/gtk-2.0/include -I/usr/lib/x86_64-linux-gnu/gtk-2.0/include -I/usr/include/glib-2.0 -I/usr/include/gtk-2.0 -I/usr/include/atk-1.0 -I/usr/include/gdk-pixbuf-2.0 -I/usr/include/cairo -I/usr/include/pango-1.0 -I/usr/lib64/glib-2.0/include -I/usr/lib/x86_64-linux-gnu/glib-2.0/include -I/usr/lib64/gtk-2.0/include -I/usr/lib/x86_64-linux-gnu/gtk-2.0/include -I/usr/include/qt5 -I/usr/include/qt5/QtSvg -I/usr/include/qt5/QtOpenGL -I/usr/include/qt5/QtWidgets -I/usr/include/qt5/QtSql -I/usr/include/qt5/QtXml -I/usr/include/qt5/QtNetwork -I/usr/include/qt5/QtGui -I/usr/include/qt5/QtCore -I. -I. -o videoframe.o src/video/videoframe.cpp
src/video/videoframe.cpp: In member function ‘bool VideoFrame::convertToRGB24(QSize)’:
src/video/videoframe.cpp:119:35: error: ‘av_free’ was not declared in this scope
         av_free(frameRGB24->opaque);
                                   ^
src/video/videoframe.cpp:120:34: error: ‘av_frame_unref’ was not declared in this scope
         av_frame_unref(frameRGB24);
                                  ^
src/video/videoframe.cpp:121:34: error: ‘av_frame_free’ was not declared in this scope
         av_frame_free(&frameRGB24);
                                  ^
src/video/videoframe.cpp:124:31: error: ‘av_frame_alloc’ was not declared in this scope
     frameRGB24=av_frame_alloc();
                               ^
src/video/videoframe.cpp:131:105: error: ‘av_malloc’ was not declared in this scope
     uint8_t* buf = (uint8_t*)av_malloc(avpicture_get_size(AV_PIX_FMT_RGB24, size.width(), size.height()));
                                                                                                         ^
src/video/videoframe.cpp:135:34: error: ‘av_frame_free’ was not declared in this scope
         av_frame_free(&frameRGB24);
                                  ^
src/video/videoframe.cpp: In member function ‘bool VideoFrame::convertToYUV420()’:
src/video/videoframe.cpp:180:32: error: ‘av_frame_alloc’ was not declared in this scope
     frameYUV420=av_frame_alloc();
                                ^
src/video/videoframe.cpp:187:91: error: ‘av_malloc’ was not declared in this scope
     uint8_t* buf = (uint8_t*)av_malloc(avpicture_get_size(AV_PIX_FMT_RGB24, width, height));
                                                                                           ^
src/video/videoframe.cpp:191:35: error: ‘av_frame_free’ was not declared in this scope
         av_frame_free(&frameYUV420);
                                   ^
src/video/videoframe.cpp: In member function ‘void VideoFrame::releaseFrameLockless()’:
src/video/videoframe.cpp:220:35: error: ‘av_free’ was not declared in this scope
         av_free(frameOther->opaque);
                                   ^
src/video/videoframe.cpp:221:34: error: ‘av_frame_unref’ was not declared in this scope
         av_frame_unref(frameOther);
                                  ^
src/video/videoframe.cpp:222:34: error: ‘av_frame_free’ was not declared in this scope
         av_frame_free(&frameOther);
                                  ^
src/video/videoframe.cpp:226:36: error: ‘av_free’ was not declared in this scope
         av_free(frameYUV420->opaque);
                                    ^
src/video/videoframe.cpp:227:35: error: ‘av_frame_unref’ was not declared in this scope
         av_frame_unref(frameYUV420);
                                   ^
src/video/videoframe.cpp:228:35: error: ‘av_frame_free’ was not declared in this scope
         av_frame_free(&frameYUV420);
                                   ^
src/video/videoframe.cpp:232:35: error: ‘av_free’ was not declared in this scope
         av_free(frameRGB24->opaque);
                                   ^
src/video/videoframe.cpp:233:34: error: ‘av_frame_unref’ was not declared in this scope
         av_frame_unref(frameRGB24);
                                  ^
src/video/videoframe.cpp:234:34: error: ‘av_frame_free’ was not declared in this scope
         av_frame_free(&frameRGB24);
                                  ^
make: *** [videoframe.o] Fehler 1

```

I have Ubuntu 14.04 installed.

The functions `av_free()`, `av_frame_unref`()``,`av_frame_free()`and`av_frame_free()``` seem to be undefined.

Does anybody know how to fix it?
This may be an issue with libav and libffmpeg.

**Comment by linux-modder** [comment]
apt-get policy ffmpeg* shows?

**Comment by zetok** [comment]
@mwuttke97 https://github.com/tux3/qTox/blob/master/INSTALL.md#dependencies ← as you see, only **ffmpeg** is supported. You need it for qTox to compile. .. and no, libav is not enough, nor it will be.

If that's all please close the issue.

**Comment by ghost** [comment]
@linux-modder 

```
apt-cache policy ffmpeg
ffmpeg:
  Installed:        (none)
  Candidate:        (none)
  Version table:
```

fmpeg is **not available** for Ubuntu 14.04 LTS.

I installed ffmpeg from source but this led to this error:

```
/usr/bin/ld: //usr/local/lib/libavdevice.a(xcbgrab.o): undefined reference to symbol 'xcb_setup_pixmap_formats_length'
//usr/lib/x86_64-linux-gnu/libxcb.so.1: error adding symbols: DSO missing from command line
collect2: error: ld returned 1 exit status
make: *** [qtox] Fehler 1
```

@zetok 
I think, this issue should not be closed yet, because this error is for all people with Ubuntu 14.04 LTS.
Is Ubuntu 14.04 LTS a not supported OS up to now?

I think, this really need something like a walk-around.

@any-ubuntu-14.04-pro:
I really don't know how to fix it. I had been trying it for hours but I have not succeeded yet.
Could someone help, please? :smile:

**Comment by ghost** [comment]
Ok, its done.

Here is, how to install ffmpeg:
1. Download ffmpeg and `cd` into it
2. `configure --enable-shared`
3. `make -j$(nproc)`
4. `sudo make install` (_Now, qTox is able to be compiled_)
5. `sudo ldconfig` (**important**)

That's all!

(We may add this to the [compiling instruductions to the required libraries](https://github.com/tux3/qTox/blob/master/INSTALL.md#gcc-qt-ffmpeg-opanal-soft-and-qrcode))
